### DISCUSSION TITLE
REST endpoint calling with default method

### DISCUSSION THREAD

**Discussion by FXschwartz** [serverpod/serverpod#2837] [Ideas]
I've been slowly migrating things off of Firebase Firestore with Firebase Functions and using a Serverpod backend.

One thing I've had to do is create new Firebase Functions with event triggers for Create, Update, and Delete on collection documents and in the functions I do a REST call to the Serverpod backend to keep the Postgres DB in sync.

The current way to call Serverpod endpoints via REST is to put the method name in the request body
```
"method": "onCreate",
```

I'd like to be able to call endpoints more traditionally, or at the very least maybe if the `method` parameter is missing from the request body Serverpod will invoke the first method in the Endpoint class by default?

This would allow me to setup an Eventarc trigger in GCP as an internal HTTP trigger, and I wouldn't have to create a Firebase function for each event type.

This is obviously all temporary and I know Serverpod is mainly focused on RPC instead of REST. But I do think there will always be cases where cross server communication or where REST is the only option.

**Comment by Isakdl** [comment]
Serverpod also support calling `/endpointName/methodName` and omitting the method param from the body.

**Reply by FXschwartz** [reply]
You are correct, that works! I had no idea we could call them like that. Thanks so much!
